xarray/tests/test_dataset.py

```diff
@@ -4635,14 +4649,19 @@ def test_to_and_from_dict(self) -> None:
         expected_no_data["coords"]["t"].update({"dtype": endiantype, "shape": (10,)})
         expected_no_data["data_vars"]["a"].update({"dtype": "float64", "shape": (10,)})
         expected_no_data["data_vars"]["b"].update({"dtype": "float64", "shape": (10,)})
-        actual_no_data = ds.to_dict(data=False)
+        actual_no_data = ds.to_dict(data=False, encoding=encoding)
         assert expected_no_data == actual_no_data
 
         # verify coords are included roundtrip
         expected_ds = ds.set_coords("b")
-        actual2 = Dataset.from_dict(expected_ds.to_dict())
+        actual2 = Dataset.from_dict(expected_ds.to_dict(data=data, encoding=encoding))
 
         assert_identical(expected_ds, actual2)
+        if encoding:
+            assert set(expected_ds.variables) == set(actual2.variables)
+            for vvh in ds.variables:
+                vv = str(vvh)
```

Comment:
using vvh results in a mypy error. not sure there's a better way to solve this, LMK if so.

Reply:
I did not check the mypy results of the precious implementation but I assume it is because you are reusing the "vv" variable and not because the variables are hashable.

Reply:
Still very new to mypy... Changing the previous `vv` (str type) variable name was the way to solve this. I tried deleting `vv` between uses but mypy didnt like that. So I changed the first `vv` to `vvs` and removed `vvh`.